src/main/java/nextstep/courses/service/SessionService.java

```diff
@@ -0,0 +1,33 @@
+package nextstep.courses.service;
+
+import java.util.Optional;
+import nextstep.courses.domain.EnrollmentRepository;
+import nextstep.courses.domain.SessionRepository;
+import nextstep.courses.domain.Sessions;
+import nextstep.courses.dto.SessionDTO;
+import nextstep.users.domain.NsUser;
+
+public class SessionService {
+
+    private final SessionRepository sessionRepository;
+    private final EnrollmentRepository enrollmentRepository;
+
+    public SessionService(SessionRepository sessionRepository, EnrollmentRepository enrollmentRepository) {
+        this.sessionRepository = sessionRepository;
+        this.enrollmentRepository = enrollmentRepository;
+    }
+
+    public Optional<Sessions> sessions(Long courseId){
+        Optional<Sessions> resultSessions = sessionRepository.findByCourseId(courseId);
+        if(resultSessions.isPresent()){
+            Sessions sessions = resultSessions.get();
+            sessions.replaceAllEnrollment(enrollmentRepository::findBySessionId);
+        }
+        return sessionRepository.findByCourseId(courseId);
+    }
+
+    public void enroll(NsUser user, SessionDTO sessionDTO){
```

Comment:
수강 신청 가능 여부를 판단해야 하지 않을까?
바로 db에 저장하기보다 Session 또는 Enrollment와 같은 객체를 통해 수강 가능여부를 판단하고 수강 신청이 가능한 경우 db에 저장해야 하지 않을까?

```suggestion
    Session session = ...
    List<Student> students = ...
    Enrollment enrollment = session.enrollment(students);
    Student student = enrollment.enroll(user);
    enrollmentRepository.save(student.getUserId(), student.getSessionId());
```

위와 유사한 형태의 코드로 구현해 보는 것은 어떨까?